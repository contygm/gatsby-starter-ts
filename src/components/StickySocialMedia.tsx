import * as React from 'react';
import FaIconLink from './FaIconLink';
import {
    faMugHot,
    faGhost,
    faEnvelope,
	faChevronRight,
	faShare
} from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { useState } from 'react';

const StickySocialMedia = (props: {isVertical: boolean}) => {
	const [showMedia, setShowMedia] = useState(true);

	const verticalClass = props.isVertical ? "is-vertical" : "";

	const handleCollapse = () => {
		setShowMedia(!showMedia)
	}

	return (
		<div 
			className={`menu sticky box mt-4 ${verticalClass}`} 
		>
			<p className="menu-label">{props.isVertical? 'Share' : 'Share this post'}</p>
			<ul className={`has-text-black has-text-centered ${props.isVertical ? "" : "columns"}`}>
				{((props.isVertical && showMedia) || (!props.isVertical)) && <>
					<div className="column">
						<FontAwesomeIcon
							className="faicon-link"
							icon={faMugHot}
							size="lg"
						/>
					</div>
					<div className="column">
						<FontAwesomeIcon
							className="faicon-link"
							icon={faGhost}
							size="lg"
						/>
					</div>
					<div className="column">
						<FontAwesomeIcon
							className="faicon-link"
							icon={faEnvelope}
							size="lg"
						/>
					</div>
				</>
				}
				{(props.isVertical && showMedia) && 
					<div className="column has-text-grey-lighter pt-4 pb-2">
						<FontAwesomeIcon
							className="faicon-link"
							icon={faChevronRight}
							size="sm"
							onClick={handleCollapse}
						/>
					</div>
				}
				{(props.isVertical && !showMedia) &&
					<div className="has-text-grey-lighter">
						<FontAwesomeIcon
							className="faicon-link"
							icon={faShare}
							size="lg"
							onClick={handleCollapse}
						/>
					</div>
				}
			</ul>
			
		</div>
	)
}

export default StickySocialMedia;