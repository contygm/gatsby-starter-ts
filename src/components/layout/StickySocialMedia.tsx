import React, { useState } from 'react';
import {
    faMugHot,
    faGhost,
    faEnvelope,
    faShare,
    faChevronUp
} from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

/**
 * A mobile and web friendly social media bar. When scrolling, the bar will stick to the top
 * of the page. The isVertical mode is ideal for mobile devices. The horizontal mode is ideal
 * for placement with in the sidebar component.
 * @param {boolean} isVertical - when true the social media bar will show vertically. This state is meant for mobile devices.
 * @param {string} [classes] - css classes to customize the look of the component
 *
 * @category Components
 */
export const StickySocialMedia = (props: {
    isVertical: boolean;
    classes?: string;
}) => {
    const [showMedia, setShowMedia] = useState(false);

    const verticalClass = props.isVertical ? 'is-vertical' : 'sticky';

    const handleCollapse = () => {
        setShowMedia(!showMedia);
    };

    return (
        <div
            className={`sticky-social-wrapper ${verticalClass} ${
                props.classes ?? ''
            }`}
        >
            <p className="menu-label">
                {props.isVertical ? 'Share' : 'Share this post'}
            </p>
            <ul
                className={`sticky-social-list-wrapper ${
                    props.isVertical ? '' : 'columns'
                }`}
            >
                {((props.isVertical && showMedia) || !props.isVertical) && (
                    <>
                        <div className="column">
                            <FontAwesomeIcon
                                className="faicon-link"
                                icon={faMugHot}
                                size="lg"
                            />
                        </div>
                        <div className="column">
                            <FontAwesomeIcon
                                className="faicon-link"
                                icon={faGhost}
                                size="lg"
                            />
                        </div>
                        <div className="column">
                            <FontAwesomeIcon
                                className="faicon-link"
                                icon={faEnvelope}
                                size="lg"
                            />
                        </div>
                    </>
                )}
                {props.isVertical && showMedia && (
                    <div className="sticky-social-hidden">
                        <FontAwesomeIcon
                            className="faicon-link"
                            icon={faChevronUp}
                            size="sm"
                            onClick={handleCollapse}
                            data-testid={'sticky-social-hide'}
                        />
                    </div>
                )}
                {props.isVertical && !showMedia && (
                    <div className="sticky-social-show">
                        <FontAwesomeIcon
                            className="faicon-link"
                            icon={faShare}
                            size="lg"
                            onClick={handleCollapse}
                            data-testid={'sticky-social-show-more'}
                        />
                    </div>
                )}
            </ul>
        </div>
    );
};
