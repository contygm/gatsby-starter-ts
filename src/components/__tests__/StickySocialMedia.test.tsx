import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import StickySocialMedia from '../StickySocialMedia';

describe('Sticky Social Media bar', () => {
    it('renders correctly horizontally', () => {
        const { asFragment } = render(<StickySocialMedia isVertical={false}/>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders correctly vertically', () => {
        const { asFragment, getByTestId } = render(<StickySocialMedia isVertical={true}/>);
        expect(asFragment()).toMatchSnapshot("open vertical bar");

		fireEvent.click(getByTestId('sticky-social-hide'))
        expect(asFragment()).toMatchSnapshot("closed vertical bar");

		fireEvent.click(getByTestId('sticky-social-show-more'))
		expect(asFragment()).toMatchSnapshot("open vertical bar");
    });
});
