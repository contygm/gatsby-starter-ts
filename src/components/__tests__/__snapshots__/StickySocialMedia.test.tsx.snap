// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sticky Social Media bar renders correctly horizontally 1`] = `
<DocumentFragment>
  <div
    class="menu sticky box mt-4  "
  >
    <p
      class="menu-label"
    >
      Share this post
    </p>
    <ul
      class="has-text-black has-text-centered columns"
    >
      <div
        class="column"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-mug-hot fa-lg faicon-link"
          data-icon="mug-hot"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M88 0C74.7 0 64 10.7 64 24c0 38.9 23.4 59.4 39.1 73.1l1.1 1C120.5 112.3 128 119.9 128 136c0 13.3 10.7 24 24 24s24-10.7 24-24c0-38.9-23.4-59.4-39.1-73.1l-1.1-1C119.5 47.7 112 40.1 112 24c0-13.3-10.7-24-24-24zM32 192c-17.7 0-32 14.3-32 32V416c0 53 43 96 96 96H288c53 0 96-43 96-96h16c61.9 0 112-50.1 112-112s-50.1-112-112-112H352 32zm352 64h16c26.5 0 48 21.5 48 48s-21.5 48-48 48H384V256zM224 24c0-13.3-10.7-24-24-24s-24 10.7-24 24c0 38.9 23.4 59.4 39.1 73.1l1.1 1C232.5 112.3 240 119.9 240 136c0 13.3 10.7 24 24 24s24-10.7 24-24c0-38.9-23.4-59.4-39.1-73.1l-1.1-1C231.5 47.7 224 40.1 224 24z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="column"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-ghost fa-lg faicon-link"
          data-icon="ghost"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M50.8 452.1L19.2 477.4c-2.1 1.7-4.7 2.6-7.4 2.6C5.3 480 0 474.7 0 468.2V192C0 86 86 0 192 0S384 86 384 192V468.2c0 6.5-5.3 11.8-11.8 11.8c-2.7 0-5.3-.9-7.4-2.6l-31.6-25.3c-3.3-2.7-7.5-4.1-11.8-4.1c-5.9 0-11.5 2.8-15 7.5l-37.6 50.1c-3 4-7.8 6.4-12.8 6.4s-9.8-2.4-12.8-6.4l-38.4-51.2c-3-4-7.8-6.4-12.8-6.4s-9.8 2.4-12.8 6.4l-38.4 51.2c-3 4-7.8 6.4-12.8 6.4s-9.8-2.4-12.8-6.4L77.6 455.5c-3.6-4.7-9.1-7.5-15-7.5c-4.3 0-8.4 1.5-11.7 4.1zM160 192c0-17.7-14.3-32-32-32s-32 14.3-32 32s14.3 32 32 32s32-14.3 32-32zm96 32c17.7 0 32-14.3 32-32s-14.3-32-32-32s-32 14.3-32 32s14.3 32 32 32z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="column"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-envelope fa-lg faicon-link"
          data-icon="envelope"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M48 64C21.5 64 0 85.5 0 112c0 15.1 7.1 29.3 19.2 38.4L236.8 313.6c11.4 8.5 27 8.5 38.4 0L492.8 150.4c12.1-9.1 19.2-23.3 19.2-38.4c0-26.5-21.5-48-48-48H48zM0 176V384c0 35.3 28.7 64 64 64H448c35.3 0 64-28.7 64-64V176L294.4 339.2c-22.8 17.1-54 17.1-76.8 0L0 176z"
            fill="currentColor"
          />
        </svg>
      </div>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`Sticky Social Media bar renders correctly vertically: closed vertical bar 1`] = `
<DocumentFragment>
  <div
    class="menu sticky box mt-4 is-vertical "
  >
    <p
      class="menu-label"
    >
      Share
    </p>
    <ul
      class="has-text-black has-text-centered "
    >
      <div
        class="has-text-grey-lighter"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-share fa-lg faicon-link"
          data-icon="share"
          data-prefix="fas"
          data-testid="sticky-social-show-more"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M307 34.8c-11.5 5.1-19 16.6-19 29.2v64H176C78.8 128 0 206.8 0 304C0 417.3 81.5 467.9 100.2 478.1c2.5 1.4 5.3 1.9 8.1 1.9c10.9 0 19.7-8.9 19.7-19.7c0-7.5-4.3-14.4-9.8-19.5C108.8 431.9 96 414.4 96 384c0-53 43-96 96-96h96v64c0 12.6 7.4 24.1 19 29.2s25 3 34.4-5.4l160-144c6.7-6.1 10.6-14.7 10.6-23.8s-3.8-17.7-10.6-23.8l-160-144c-9.4-8.5-22.9-10.6-34.4-5.4z"
            fill="currentColor"
          />
        </svg>
      </div>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`Sticky Social Media bar renders correctly vertically: open vertical bar 1`] = `
<DocumentFragment>
  <div
    class="menu sticky box mt-4 is-vertical "
  >
    <p
      class="menu-label"
    >
      Share
    </p>
    <ul
      class="has-text-black has-text-centered "
    >
      <div
        class="column"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-mug-hot fa-lg faicon-link"
          data-icon="mug-hot"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M88 0C74.7 0 64 10.7 64 24c0 38.9 23.4 59.4 39.1 73.1l1.1 1C120.5 112.3 128 119.9 128 136c0 13.3 10.7 24 24 24s24-10.7 24-24c0-38.9-23.4-59.4-39.1-73.1l-1.1-1C119.5 47.7 112 40.1 112 24c0-13.3-10.7-24-24-24zM32 192c-17.7 0-32 14.3-32 32V416c0 53 43 96 96 96H288c53 0 96-43 96-96h16c61.9 0 112-50.1 112-112s-50.1-112-112-112H352 32zm352 64h16c26.5 0 48 21.5 48 48s-21.5 48-48 48H384V256zM224 24c0-13.3-10.7-24-24-24s-24 10.7-24 24c0 38.9 23.4 59.4 39.1 73.1l1.1 1C232.5 112.3 240 119.9 240 136c0 13.3 10.7 24 24 24s24-10.7 24-24c0-38.9-23.4-59.4-39.1-73.1l-1.1-1C231.5 47.7 224 40.1 224 24z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="column"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-ghost fa-lg faicon-link"
          data-icon="ghost"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M50.8 452.1L19.2 477.4c-2.1 1.7-4.7 2.6-7.4 2.6C5.3 480 0 474.7 0 468.2V192C0 86 86 0 192 0S384 86 384 192V468.2c0 6.5-5.3 11.8-11.8 11.8c-2.7 0-5.3-.9-7.4-2.6l-31.6-25.3c-3.3-2.7-7.5-4.1-11.8-4.1c-5.9 0-11.5 2.8-15 7.5l-37.6 50.1c-3 4-7.8 6.4-12.8 6.4s-9.8-2.4-12.8-6.4l-38.4-51.2c-3-4-7.8-6.4-12.8-6.4s-9.8 2.4-12.8 6.4l-38.4 51.2c-3 4-7.8 6.4-12.8 6.4s-9.8-2.4-12.8-6.4L77.6 455.5c-3.6-4.7-9.1-7.5-15-7.5c-4.3 0-8.4 1.5-11.7 4.1zM160 192c0-17.7-14.3-32-32-32s-32 14.3-32 32s14.3 32 32 32s32-14.3 32-32zm96 32c17.7 0 32-14.3 32-32s-14.3-32-32-32s-32 14.3-32 32s14.3 32 32 32z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="column"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-envelope fa-lg faicon-link"
          data-icon="envelope"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M48 64C21.5 64 0 85.5 0 112c0 15.1 7.1 29.3 19.2 38.4L236.8 313.6c11.4 8.5 27 8.5 38.4 0L492.8 150.4c12.1-9.1 19.2-23.3 19.2-38.4c0-26.5-21.5-48-48-48H48zM0 176V384c0 35.3 28.7 64 64 64H448c35.3 0 64-28.7 64-64V176L294.4 339.2c-22.8 17.1-54 17.1-76.8 0L0 176z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="column has-text-grey-lighter pt-4 pb-2"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-chevron-right fa-sm faicon-link"
          data-icon="chevron-right"
          data-prefix="fas"
          data-testid="sticky-social-hide"
          focusable="false"
          role="img"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M342.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-192 192c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L274.7 256 105.4 86.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l192 192z"
            fill="currentColor"
          />
        </svg>
      </div>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`Sticky Social Media bar renders correctly vertically: open vertical bar 2`] = `
<DocumentFragment>
  <div
    class="menu sticky box mt-4 is-vertical "
  >
    <p
      class="menu-label"
    >
      Share
    </p>
    <ul
      class="has-text-black has-text-centered "
    >
      <div
        class="column"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-mug-hot fa-lg faicon-link"
          data-icon="mug-hot"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M88 0C74.7 0 64 10.7 64 24c0 38.9 23.4 59.4 39.1 73.1l1.1 1C120.5 112.3 128 119.9 128 136c0 13.3 10.7 24 24 24s24-10.7 24-24c0-38.9-23.4-59.4-39.1-73.1l-1.1-1C119.5 47.7 112 40.1 112 24c0-13.3-10.7-24-24-24zM32 192c-17.7 0-32 14.3-32 32V416c0 53 43 96 96 96H288c53 0 96-43 96-96h16c61.9 0 112-50.1 112-112s-50.1-112-112-112H352 32zm352 64h16c26.5 0 48 21.5 48 48s-21.5 48-48 48H384V256zM224 24c0-13.3-10.7-24-24-24s-24 10.7-24 24c0 38.9 23.4 59.4 39.1 73.1l1.1 1C232.5 112.3 240 119.9 240 136c0 13.3 10.7 24 24 24s24-10.7 24-24c0-38.9-23.4-59.4-39.1-73.1l-1.1-1C231.5 47.7 224 40.1 224 24z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="column"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-ghost fa-lg faicon-link"
          data-icon="ghost"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M50.8 452.1L19.2 477.4c-2.1 1.7-4.7 2.6-7.4 2.6C5.3 480 0 474.7 0 468.2V192C0 86 86 0 192 0S384 86 384 192V468.2c0 6.5-5.3 11.8-11.8 11.8c-2.7 0-5.3-.9-7.4-2.6l-31.6-25.3c-3.3-2.7-7.5-4.1-11.8-4.1c-5.9 0-11.5 2.8-15 7.5l-37.6 50.1c-3 4-7.8 6.4-12.8 6.4s-9.8-2.4-12.8-6.4l-38.4-51.2c-3-4-7.8-6.4-12.8-6.4s-9.8 2.4-12.8 6.4l-38.4 51.2c-3 4-7.8 6.4-12.8 6.4s-9.8-2.4-12.8-6.4L77.6 455.5c-3.6-4.7-9.1-7.5-15-7.5c-4.3 0-8.4 1.5-11.7 4.1zM160 192c0-17.7-14.3-32-32-32s-32 14.3-32 32s14.3 32 32 32s32-14.3 32-32zm96 32c17.7 0 32-14.3 32-32s-14.3-32-32-32s-32 14.3-32 32s14.3 32 32 32z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="column"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-envelope fa-lg faicon-link"
          data-icon="envelope"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M48 64C21.5 64 0 85.5 0 112c0 15.1 7.1 29.3 19.2 38.4L236.8 313.6c11.4 8.5 27 8.5 38.4 0L492.8 150.4c12.1-9.1 19.2-23.3 19.2-38.4c0-26.5-21.5-48-48-48H48zM0 176V384c0 35.3 28.7 64 64 64H448c35.3 0 64-28.7 64-64V176L294.4 339.2c-22.8 17.1-54 17.1-76.8 0L0 176z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="column has-text-grey-lighter pt-4 pb-2"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-chevron-right fa-sm faicon-link"
          data-icon="chevron-right"
          data-prefix="fas"
          data-testid="sticky-social-hide"
          focusable="false"
          role="img"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M342.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-192 192c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L274.7 256 105.4 86.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l192 192z"
            fill="currentColor"
          />
        </svg>
      </div>
    </ul>
  </div>
</DocumentFragment>
`;
